packages/core/src/root.ts

```diff
@@ -147,6 +154,8 @@ export function createRemoteRoot<
       makeRemote(text, id, remoteRoot);
       texts.set(text, content);
 
+      componentsSet.add(text as any);
```

Comment:
Why is the cast needed here? Also, given that this contains both texts and components, might be better to just call the set `nodes` or something like that.